### PR TITLE
fix(mIO): allow mIO as an input to write APIs

### PR DESCRIPTION
**Pull request by dtfiedler**
Allows `mIOToken` class to be provided as an input parameter to write interaction APIs.

### CODE REVIEW

**Review by fedellen — DISMISSED** [review]

**Review by fedellen — APPROVED** [review]

File: src/common/ar-io.ts [hunk 1]
```diff
@@ -563,7 +565,7 @@ export class ArIOWritable extends ArIOReadable implements ArIOWriteContract {
 
   /**
    * @param target @type {string} The address of the account you want to transfer IO tokens to.
-   * @param qty @type {number} The amount of IO or mIO to transfer.
+   * @param qty @type {number | mIOToken} The amount of IO or mIO to transfer.
```
Comment (atticusofsparta) [inline src/common/ar-io.ts:568]: is it possible for us to put a deprecated warning that number will be unavailable in the future? or is that overkill for this?

File: tests/integration/arlocal/ar-io-contract/index.js [hunk 2]
```diff
@@ -7485,10 +7486,9 @@ var TransferToken = class {
         getInvalidAjvMessage(validateTransferToken, input, 'transferToken'),
```
Comment (atticusofsparta) [inline tests/integration/arlocal/ar-io-contract/index.js:7486]: is this stuff supposed to be in here?

Comment (dtfiedler) [reply tests/integration/arlocal/ar-io-contract/index.js:7486]: it's the newest copy of the contract source code that expects everything in mIO

File: tests/integration/ar-io-writable.test.ts [hunk 3]
```diff
@@ -49,7 +50,7 @@ describe('ArIOWriteable', () => {
   it('should successfully increase delegate stake', async () => {
     const tx = await arIO.increaseDelegateStake({
       target: gatewayAddress,
-      qty: 101,
+      qty: new IOToken(100).toMIO(),
```
Comment (kunstmusik) [inline tests/integration/ar-io-writable.test.ts:53]: would be nice if the test checked the before/after value on the gateway matches expected values after increase/decreaseDelegateStake

Comment (dtfiedler) [reply tests/integration/ar-io-writable.test.ts:55]: [45e3521](https://github.com/ar-io/ar-io-sdk/pull/92/commits/45e3521f7701fff1be0870dcf2f8af65e68d1cf8)

**Review by kunstmusik — APPROVED** [review]

**Pull request merged**